server/src/apub/extensions/signatures.rs

```diff
@@ -36,56 +37,41 @@ pub fn generate_actor_keypair() -> Result<Keypair, Error> {
 }
 
 /// Signs request headers with the given keypair.
-pub fn sign(request: &Builder, actor: &dyn ActorType) -> Result<String, Error> {
+pub async fn sign(
+  request: ClientRequest,
+  actor: &dyn ActorType,
+  activity: String,
+) -> Result<DigestClient<String>, LemmyError> {
   let signing_key_id = format!("{}#main-key", actor.actor_id());
+  let private_key = actor.private_key();
 
-  let headers = request
-    .headers_ref()
-    .unwrap()
-    .iter()
-    .map(|h| -> Result<(String, String), Error> {
-      Ok((h.0.as_str().to_owned(), h.1.to_str()?.to_owned()))
-    })
-    .collect::<Result<BTreeMap<String, String>, Error>>()?;
+  let digest_client = request
+    .signature_with_digest(
```

Comment:
So I also added digest headers to your requests for payload verification, however, there's nothing actually verifying the digests on the other end. It should be pretty simple to add http-signature-normalization-actix's digest verify middleware, but I didn't want to do that in this PR